File: mpp/LabMppCurseWPF/ServicesInterfaces/IServer.cs
using System;
using System.Collections.Generic;
using System.Data;
using CurseMPP.Models;
using LabMppCurseWPF.Models;

namespace ServicesInterfaces
{
	public interface IServer
	{
		void Login(Personal personal, IObserver client);
		Cursa[] GetAllCurse(Personal personal);
		Destinatie[] GetAllDestinatii();
		Loc[] GetLocuri(int idCursa);
		Cursa FindCursa(DateTime data, Destinatie destinatie);
		void AdaugaRezervare(int idCursa, string nume, string prenume, int nrLocuri);
	}
}

File: mpp/LabMPPCurse/ServiceInterfaces/src/main/java/services/IServer.java
package services;

import Models.Cursa.Cursa;
import Models.Destinatie.Destinatie;
import Models.Loc;
import Models.Personal.Personal;
import Models.Rezervare;

public interface IServer {
     void login(Personal personal, IObserver client) throws ServerException;
     Destinatie[] getDestinatii() throws  ServerException;
     Cursa[] getCurse() throws ServerException;
     Loc[] getLocuri(Cursa c) throws ServerException;
     void addRezervare(int idCursa, String nume, String prenume, int nrLocuri) throws ServerException;
//     void sendMessage(Message message) throws ChatException;
//     void logout(User user, IChatObserver client) throws ChatException;
//     User[] getLoggedFriends(User user) throws ChatException;
}
